src/request/import-file/ImportFileApi.js

```diff
@@ -1,136 +1,217 @@
+        const $cancelLink = ($appName.parentNode);
+        $cancelLink.classList.remove('display-none');
+        $cancelLink.addEventListener('click', () => window.close());
+
+        this.run();
     }
 
     /**
-     * @returns {{passphraseInput: PassphraseInput, pinInput: PinInput}}
+     * @returns {{passphraseBox: PassphraseBox, passphraseSetterBox: PassphraseSetterBox}}
      */
     _makeView() {
-        // Pages
-        /** @type {HTMLElement} */
-        const $importPage = (document.getElementById(ImportFileApi.Pages.FILE_IMPORT));
-        /** @type {HTMLElement} */
-        const $passphrasePage = (document.getElementById(ImportFileApi.Pages.ENTER_PASSPHRASE));
-        /** @type {HTMLElement} */
-        const $pinPage = (document.getElementById(ImportFileApi.Pages.ENTER_PIN));
-
         // Containers
         /** @type {HTMLDivElement} */
-        const $fileImportComponent = ($importPage.querySelector('#file-import-component'));
+        const $fileImport = (document.querySelector('.file-import'));
         /** @type {HTMLFormElement} */
-        const $passphraseInput = ($passphrasePage.querySelector('#passphrase-component'));
-        /** @type {HTMLDivElement} */
-        const $pinInput = ($pinPage.querySelector('#pin-component'));
+        const $passphraseBox = (document.querySelector('.passphrase-box'));
+        /** @type {HTMLFormElement} */
+        const $passphraseSetterBox = (document.querySelector('.passphrase-setter-box'));
+        /** @type {HTMLElement} */
+        const backArrow = (document.querySelector(`#${ImportFileApi.Pages.ENTER_PASSPHRASE} .page-header-back-button`));
 
         // Components
-        const fileImportComponent = new FileImport($fileImportComponent);
-        const passphraseInput = new PassphraseInput($passphraseInput);
-        const pinInput = new PinInput($pinInput);
+        const fileImport = new FileImport($fileImport);
+        const passphraseBox = new PassphraseBox($passphraseBox, { buttonI18nTag: 'passphrasebox-log-in' });
+        const passphraseSetterBox = new PassphraseSetterBox($passphraseSetterBox);
 
         // Events
-        fileImportComponent.on(FileImport.Events.IMPORT, this._onFileImported.bind(this));
-        passphraseInput.on(PassphraseInput.Events.PASSPHRASE_ENTERED, this._onPassphraseEntered.bind(this));
-        pinInput.on(PinInput.Events.PIN_ENTERED, this._onPinEntered.bind(this));
+        fileImport.on(FileImport.Events.IMPORT, this._onFileImported.bind(this));
+        passphraseBox.on(PassphraseBox.Events.SUBMIT, this._onPassphraseEntered.bind(this));
+        passphraseBox.on(PassphraseBox.Events.CANCEL, () => this._goToImportPage());
+        passphraseSetterBox.on(PassphraseSetterBox.Events.SUBMIT, this._onPassphraseEntered.bind(this));
+        passphraseSetterBox.on(PassphraseSetterBox.Events.SKIP, () => this._onPassphraseEntered(null));
+        backArrow.addEventListener('click', () => this._goToImportPage());
```

Comment:
Is this a special case or can we remove it after merging #84 ?

Reply:
I think it's covered by #84 and must be removed.